### ISSUE TITLE
List of custom data type not persisted in app state

### ISSUE THREAD

**Issue by pyro32** [FlutterFlow/flutterflow-issues#2391]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

List of custom data type not persisted in app state. 

### Expected Behavior

Elements added to a list of a custom data type are not persisted when using the configuration UI, when testing in debug mode, or running the application.

### Steps to Reproduce

1. Create new custom data type
2. Create new persisted app state variable that is a list of the custom data type
3. Add an item to the default value in the Flutterflow app state UI
4. Leave app state UI
5. Reenter app state UI
6. Observe default value not persisted

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

IT4gj8ncx891rt9G1s/AL+9ajj46KWYiRL00r9pEGEshfJDsBLQDds/8QElLYteUYwtIGGaj+HIE/sLRktrpGMc7BE6tUb5Az75XQRWXTmKtMLbUDqjtPEElJ/pXIW6+0J6wkBRTPvt1LEY872eYffCdG3qCf9qOYwx5e6fDbOY=

### Context

This is blocking my ability to persist data that is loaded into the app

### Visual documentation

![image](https://github.com/FlutterFlow/flutterflow-issues/assets/39568874/5a842f96-9ea1-4ee1-9827-33b2fe529377)

### Environment

```markdown
- FlutterFlow version: v4.1
- Platform: Web
- Browser name and version: Chrome, Version 121.0.6167.189 (Official Build) (64-bit)
- Operating system and version affected: Windows 11 Home, 22H2
```

**Comment by hariprasadms** [comment]
Hi @pyro32 -  Thanks for bringing up the issue. I've tested the steps you mentioned above. Everything seems to be working fine on my end, and I couldn't replicate the behavior you're experiencing. Could you please confirm if the issue still persists on your end?

**Comment by pyro32** [comment]
I'm wondering if this was failing silently because of an appstate
storage quota. I'm now seeing this exception in the console for my other
persisted variables that were working fine before. Wondering if maybe I was
hitting the (unknown) storage quota on one  of the earlier persisted
variables and then the following variables were not attempted to be
persisted and silently failing. I'll try to debug this later today. Can you
share any information about storage quota limits? Could we improve the
error messages around this if this is actually a quota issue?

**Comment by bozzoleandro** [comment]
Hi @pyro32, are you still experiencing this issue? Thanks!

**Comment by pyro32** [comment]
Hey I was waiting to hear about SharedPerferences storage limits but I ended up implementing a Hive database to store my serialized objects, then load them back into Appstate when needed. I would recommend adding some error handling if there are limits here that are been hit when persisting Appstate variables since intermittently hitting these and having them fail silently took a while to diagnose.

**Comment by bozzoleandro** [comment]
Hi @pyro32, I've been taking a look at the issue, and I think that you need to tab on the dropdown and you'll see what you added.

Please let me know if that works for you.

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/5e6d6018-7434-4082-8745-2df0a2182a25/605b7a25-864d-4853-a856-fd8c5deca1ae?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS81ZTZkNjAxOC03NDM0LTQwODItODc0NS0yZGYwYTIxODJhMjUvNjA1YjdhMjUtODY0ZC00ODUzLWE4NTYtZmQ4YzVkZWNhMWFlIiwiaWF0IjoxNzEzMjk2MzEyLCJleHAiOjMzMjgzODU2MzEyfQ.-q8Jh_R8BfN0CObFKPyLXQWlm0OlU9ygddnYYeKvgiI " alt="image.png" width="1100" height="351" />

<img src="https://uploads.linear.app/3241089d-736a-4422-900b-051e72accdbe/05812944-b2fd-48bf-8d43-fb841c61cdaa/7dc4b67f-80df-45c9-a118-2367459f177c?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzMyNDEwODlkLTczNmEtNDQyMi05MDBiLTA1MWU3MmFjY2RiZS8wNTgxMjk0NC1iMmZkLTQ4YmYtOGQ0My1mYjg0MWM2MWNkYWEvN2RjNGI2N2YtODBkZi00NWM5LWExMTgtMjM2NzQ1OWYxNzdjIiwiaWF0IjoxNzEzMjk2MzEyLCJleHAiOjMzMjgzODU2MzEyfQ.gxASY3YwqxSu_j8MaX2SUVEbOLxuaTJXWuSV9b3ZApY " alt="image.png" width="2140" height="672" />

**Comment by pyro32** [comment]
Hey there - that's definitely not the issue. Other app state variable I created worked just fine. I'm 99% sure the issue is I exceeded the limit for what could be persisted using SharedPreferences. I saw an error in console while debugging on web. It was working fine until I magically hit some data threshold then it started failing without any indication in the flutterflow UI that there was a problem. Might be good to add some error reporting in the UI somehow or at least warn users there is a limit for how much data can be persisted using this mechanism.
